spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfigurationTests.java

```diff
@@ -51,118 +47,102 @@
 import org.springframework.cloud.skipper.client.SkipperClient;
 import org.springframework.cloud.task.configuration.SimpleTaskAutoConfiguration;
 import org.springframework.cloud.task.repository.TaskRepository;
-import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.context.annotation.Bean;
-import org.springframework.core.env.ConfigurableEnvironment;
-import org.springframework.core.env.MapPropertySource;
-import org.springframework.core.env.MutablePropertySources;
-import org.springframework.core.env.StandardEnvironment;
+import org.springframework.core.NestedExceptionUtils;
 import org.springframework.hateoas.config.EnableHypermediaSupport;
 import org.springframework.security.authentication.AuthenticationManager;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
 
 /**
  * @author Glenn Renfro
  * @author Ilayaperumal Gopinathan
  * @author Gunnar Hillert
+ * @author Michael Wirth
  */
 public class DataFlowServerConfigurationTests {
 
-	private AnnotationConfigApplicationContext context;
+	private ApplicationContextRunner contextRunner;
 
-	private ConfigurableEnvironment environment;
-
-	private MutablePropertySources propertySources;
-
-	@Before
+	@BeforeEach
 	public void setup() {
-		context = new AnnotationConfigApplicationContext();
-		context.setId("testDataFlowConfig");
-		context.register(DataFlowServerConfigurationTests.TestConfiguration.class,
-				SecurityAutoConfiguration.class, DataFlowServerAutoConfiguration.class,
-				DataFlowControllerAutoConfiguration.class, DataSourceAutoConfiguration.class,
-				DataFlowServerConfiguration.class, PropertyPlaceholderAutoConfiguration.class,
-				RestTemplateAutoConfiguration.class, HibernateJpaAutoConfiguration.class, WebConfiguration.class,
-				SchedulerConfiguration.class, JacksonAutoConfiguration.class, SimpleTaskAutoConfiguration.class,
-				ResourceLoadingAutoConfiguration.class, ComposedTaskRunnerConfigurationProperties.class);
-		environment = new StandardEnvironment();
-		propertySources = environment.getPropertySources();
-	}
-
-	@After
-	public void teardown() {
-		if (context != null) {
-			context.close();
-		}
+		contextRunner = new ApplicationContextRunner()
```

Comment:
ℹ️ you will notice in my example below that the ACR is defined as member var rather than setup. The reason is that it's API always returns a new instance (it is designed as such).

Reply:
I updated the orginal test

Reply:
👍🏻 you are more than welcome. It is definitely one of my favorites.